### PR TITLE
add command_check to common.sh

### PR DESCRIPTION
**Pull request by grufwub**
Provides an alternative method of checking for previous installs (e.g. go or nodejs) by looking both in PATH, and at the expected location. Have converted the cloudcli doclt, docker and ruby scripts to use this new function

### CODE REVIEW

File: pengwin-setup.d/common.sh [hunk 1]
```diff
@@ -52,6 +52,21 @@ echo "Removing unnecessary packages."
 sudo apt-get autoremove -y
 }
 
+function command_check() {
+# Usage: command_check <EXPECTED PATH> <ARGS (if any)>
```
Comment (crramirez) [inline pengwin-setup.d/common.sh]: Please indent with 2 spaces

File: pengwin-setup.d/common.sh [hunk 2]
```diff
@@ -52,6 +52,21 @@ echo "Removing unnecessary packages."
...
+local execname=$(echo "$1" | sed -e "s|^.*\/||g")
+if "$execname" "$2" > /dev/null 2>&1 ; then
```
Comment (crramirez) [inline pengwin-setup.d/common.sh]: Please put it in parenthesis for improving the reading

File: pengwin-setup.d/common.sh [hunk 3]
```diff
@@ -52,6 +52,21 @@ echo "Removing unnecessary packages."
...
+	echo "Executable $execname in PATH"
```
Comment (crramirez) [inline pengwin-setup.d/common.sh]: Please change tabs by 2 spaces

File: pengwin-setup.d/docker.sh [hunk 4]
```diff
@@ -18,13 +18,18 @@ function docker_install_build_relay() {
...
       echo "Downloading Go using wget."
       wget -c "https://dl.google.com/go/go${GOVERSION}.linux-$(dpkg --print-architecture).tar.gz"
       tar -xzf go*.tar.gz
 
       export GOROOT=$(pwd)/go
       export PATH="${GOROOT}/bin:$PATH"
+    else
+      # Whether it was just installed previously, or right now,
+      # makes sure to set correct env variables
+      export GOROOT=/usr/local/go
```
Comment (crramirez) [inline pengwin-setup.d/docker.sh]: What happens if Go was installed via **apt**? it will be in another path

Comment (grufwub) [reply pengwin-setup.d/docker.sh]: modified this in latest commit. there's now 3 different return values from the command_check so these variables are only set if it's a fresh go install performed via pengwin-setup

File: pengwin-setup.d/common.sh (hunk 3 above)
Comment (grufwub) [reply pengwin-setup.d/common.sh]: should be resolved now, will make another PR once this one has gone through fixing up the rest of the spacing inconsistencies throughout the file with other functions

File: pengwin-setup.d/common.sh (hunk 2 above)
Comment (grufwub) [reply pengwin-setup.d/common.sh]: done :+1:

File: pengwin-setup.d/common.sh (hunk 1 above)
Comment (grufwub) [reply pengwin-setup.d/common.sh]: should be resolved now, will make another PR once this one has gone through fixing up the rest of the spacing inconsistencies throughout the file with other functions

**Pull request merged**